tests/0139-offset_validation_mock.c

```diff
@@ -165,10 +166,9 @@ static void do_test_ssl_error_retried(void) {
                                  "bootstrap.servers", bootstraps,
                                  "batch.num.messages", "1", NULL);
 
-        /* Make OffsetForLeaderEpoch fail with the _SSL error */
-        rd_kafka_mock_push_request_errors(mcluster,
-                                          RD_KAFKAP_OffsetForLeaderEpoch, 1,
-                                          RD_KAFKA_RESP_ERR__SSL);
+        /* Make OffsetForLeaderEpoch fail with the corresponding error code */
+        rd_kafka_mock_push_request_errors(
+            mcluster, RD_KAFKAP_OffsetForLeaderEpoch, 1, err);
```

Comment:
We ideally should not be sending local errors via the mock broker response, it's not what happens, right?

Reply:
Yeah, but the effect to test would be the same, the SSL error was local too and this seems a simple way to reproduce them.
Other option is a unit test but more complex to set up.

Reply:
Sure, go ahead and merge this, looks good to me